File: 1. Study/0527 - 완전탐색(블루투포스)/mojaeya/3_멘토링.js
let arr = [
  [3, 4, 1, 2],
  [4, 3, 2, 1],
  [3, 1, 4, 2],
];
console.log(solution(4, 3, arr));

function solution(N, M, test) {
  let answer = 0;
  //   let tmp = [];
  for (let i = 1; i <= N; i++) {
    for (let j = 1; j <= N; j++) {
      let count = 0;
      for (let k = 0; k < M; k++) {
        let pi = (pj = 0);
        for (let s = 0; s < N; s++) {
          if (test[k][s] === i) pi = s;
          if (test[k][s] === j) pj = s;
        }
        if (pi < pj) count++;
      }
      if (count === M) {
        // tmp.push([i, j]);
        answer++;
      }
    }
  }
  //   console.log(tmp);
  return answer;
}

File: 1. Study/0527 - 완전탐색(블루투포스)/kimshin/3.멘토링.py
def sol(test):
    answer=0
    m=len(test)
    n=len(test[0])

    for i in range(1,n+1):
        for j in range(1,n+1):
            # print(i,j)
            cnt=0
            for k in range(m):
                pi=pj=0
                for s in range(n):
                    if test[k][s] == i:
                        pi=s
                    if test[k][s] == j:
                        pj = s
                if pi < pj:
                    cnt+=1
            if cnt ==m:
                answer+=1
    return answer



arr=[[3,4,1,2],[4,3,2,1],[3,1,4,2]]
print(sol(arr))